.github/workflows/CI.yml

```diff
@@ -0,0 +1,36 @@
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+name: Continuous integration
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: install system deps
+        run: sudo apt install libdbus-1-dev libparted-dev
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - name: install system deps
+        run: sudo apt install libdbus-1-dev libparted-dev
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          components: clippy
+      - uses: actions-rs/cargo@v1
```

Comment:
use [`clippy-check@v1`](https://github.com/actions-rs/clippy-check)

Reply:
see my [comment above](https://github.com/pop-os/upgrade/pull/144#issuecomment-811702277).

clippy-check expects a github token to be set up, i believe. but it *is* my preferred option

Reply:
... which i suddenly see [is created by default](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret).

i'll update the pull request